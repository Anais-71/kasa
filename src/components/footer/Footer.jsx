import logoWhite from '../../assets/logo_white.svg'
import './footer.css'

function Footer() {
    return (
        <div className='footer'>
            <img src={logoWhite} alt='Kasa' className='logo' />
            <div className='copyrights'>
                <p>Â© 2020 Kasa. All rights reserved</p>
            </div>
        </div>
    )
}

export default Footer
