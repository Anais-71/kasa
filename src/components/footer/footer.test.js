import { render, screen } from '@testing-library/react';
import Footer from './Footer';

describe('Given I am on a page', () => {
  beforeEach(() => {
    render(<Footer />);
  });

  test('When the footer component is being displayed, Then the white logo should be present', () => {
    const logo = screen.getByAltText('Kasa');
    expect(logo).toBeInTheDocument();
  });  

  test('When the footer component is being displayed, Then the copyrights should be present', () => {
    const copyrights = screen.getByText(/All rights reserved/i);
    expect(copyrights).toBeInTheDocument();
  });

  test('The footer page matches the snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});