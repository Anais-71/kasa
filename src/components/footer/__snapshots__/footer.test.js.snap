// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given I am on a page The footer page matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <img
      alt="Kasa"
      class="logo"
      src="logo_white.svg"
    />
    <div
      class="copyrights"
    >
      <p>
        Â© 2020 Kasa. All rights reserved
      </p>
    </div>
  </div>
</DocumentFragment>
`;
